src/commands/availability/availability.js

```diff
@@ -0,0 +1,368 @@
+// import ApplicationCommandOptionType if you need types of options
+// import PermissionFlagsBits if you need permissions
+const { SlashCommandBuilder, EmbedBuilder } = require('discord.js');
+
+const { ScheduleError, parseDaysList } = require('../../utils/schedule');
+const dayjs = require('dayjs');
+const {
+    createAvailability, createUnavailability, saveUnavailability, saveAvailability, loadAvailability
+} = require('../../utils/availability');
+const path = './src/savedAvailability.json';
+
+// Availability Data Schema
+// {
+//   "id": {
+//     "userId": "1223334444",
+//     "userTag": "servermember",
+//     "available": {
+//       "from": "2024-05-20T14:00:00.000Z",
+//       "to": "2024-08-09T22:00:00.000Z",
+//       "days": [
+//         "monday",
+//         "tuesday"
+//       ]
+//     },
+//     "unavailable": [
+//       {
+//         "from": "2024-07-03T05:00:00.000Z",
+//         "to": "2024-07-06T04:59:00.000Z",
+//         "reason": "Holiday"
+//       },
+//       {
+//         "from": "2024-08-01T20:00:00.000Z",
+//         "to": "2024-08-02T17:30:00.000Z"
+//       }
+//     ]
+//   }
+// }
```

Comment:
This is good for testing purposes, but can be removed now since the json should have this data structure